src/extension.ts

```diff
@@ -269,10 +269,10 @@ class MarkdownTocTools {
         let insertSpaces    = <boolean> workspace.getConfiguration("[markdown]")["editor.insertSpaces"];
         
         if(lineEnding === undefined || lineEnding === null || lineEnding === "auto") {
-            // Default back to "\n" in case of VSCode update 1.29.0,
-            // where `Files > EOL` was set to the string value "auto".
+            // Use OS default line-endings where `Files > EOL` was set to the string value "auto".
+            // "eol=auto" was introduced in VSCode update 1.29.0.
             // See https://github.com/AlanWalk/markdown-toc/issues/65 
-            lineEnding = "\n"; 
+            lineEnding = require('os').EOL; 
```

Comment:
Should probably change this to an import at the top of the file

```javascript
import { EOL } from 'os';
```

This also allows for a change on line 249 in the `insertAnchor` method where it builds the `text` variable.

This

```javascript
let text = [ '<a id="markdown-', name, '" name="', name, '"></a>\n' ];
```

Can be changed to
```javascript
let text = [ '<a id="markdown-', name, '" name="', name, '"></a>', EOL ];
```